### PR TITLE
Add Withdrawal Amount and Index validation

### PR DESCRIPTION
**Pull request by LukaszRozmej**
Adds additional validation

### CODE REVIEW

File: src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs [hunk 1]
```diff
@@ -155,37 +156,47 @@ public bool ValidateProcessedBlock(Block processedBlock, TxReceipt[] receipts, B
...
             if (!ValidateWithdrawalsHashMatches(block, out Keccak withdrawalsRoot))
             {
-                error = $"Withdrawals root hash mismatch in block {block.ToString(Block.Format.FullHashAndNumber)}: expected {block.Header.WithdrawalsRoot}, got {withdrawalsRoot}";
-                if (_logger.IsWarn) _logger.Warn($"Withdrawals root hash mismatch in block {block.ToString(Block.Format.FullHashAndNumber)}: expected {block.Header.WithdrawalsRoot}, got {withdrawalsRoot}");
+                return Fail(_logger, error = $"Withdrawals root hash mismatch in block {block.ToString(Block.Format.FullHashAndNumber)}: expected {block.Header.WithdrawalsRoot}, got {withdrawalsRoot}");
+            }
 
-                return false;
+            // We should initialize it from parent?
+            ulong expectedIndex = block.Withdrawals.FirstOrDefault()?.Index ?? 0;
```
Comment (MarekM25) [inline src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs:183]: we're starting from 0 according to EIP
"a monotonically increasing index, starting from 0," Not sure if FirstOrDefault is needed

Comment (MarekM25) [inline src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs:183]: If we decide to validate indexes, we should validate amount != 0 for sure

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs:183]: Yes, but starting from 0 is between blocks. In a block it starts at some index (that is index in last block +1)

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs:183]: We do [GetPooledTransactionsMessage](https://github.com/NethermindEth/nethermind/pull/5123/files#diff-e3b880e83ad43a7ae2bb98490f625d74598e445fab37ea418ea354fcb633de51R187)

Comment (MarekM25) [reply src/Nethermind/Nethermind.Consensus/Validators/BlockValidator.cs:183]: yes, agreed. I observed it on devnet

**Comment by MarekM25** [comment]
closing - we all agreed that we don't need this validation

**Pull request closed without merging**